.github/workflows/main.yml

```diff
@@ -56,6 +56,25 @@ jobs:
       platform: ${{ matrix.platform }}
       force-minimum-dependencies: ${{ matrix.force-minimum-dependencies }}
       run-slow: ${{ inputs.run-slow || false }}
+  check_depends:
+    name: Run dependency checks
+    uses: ./.github/workflows/check_depends.yml
+    # https://wildwolf.name/github-actions-how-to-avoid-running-the-same-workflow-multiple-times/
+    if: >
+      github.event_name != 'pull_request'
+      || github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name
+    strategy:
+      matrix:
+        # We run on just one release of Python and one OS, as they should be all alike.
```

Comment:
Building off my other suggestion of giving the tox environment a more generic name, I would actually suggest just running this check from the normal `tests.yml` workflow. It might be useful for more than just ensuring all the dependencies are installed, and whatever other stuff it does check could conceivably vary from one Python version to another. (And actually, dependencies can be constrained to specific Python versions so there's probably some value to running the check on all Python versions just because of that.)

So I'd suggest just putting a separate step in `tests.yml` with a separate `tox` invocation that uses the new environment.